app/assets/javascripts/discourse/app/models/topic.js

```diff
@@ -467,7 +467,12 @@ const Topic = RestModel.extend({
           "details.can_permanently_delete":
             this.siteSettings.can_permanently_delete && deleted_by.admin,
         });
-        if (!deleted_by.staff) {
+        if (
+          !deleted_by.staff &&
+          !deleted_by.groups.some(
+            (group) => group.name === this.category.reviewable_by_group_name
```

Comment:
Isn't `reviewable_by_group_name` leaking names of private groups?

Reply:
No. Since the we have a guard of `can_see_serialized_category?` it will return undefined for non group moderators.